src/precomputed_quantities.jl

```diff
@@ -16,10 +16,13 @@ end
 function precomputed_quantities!(Y, p, t, colidx)
     ᶜuₕ = Y.c.uₕ
     ᶠw = Y.f.w
-    (; ᶜuvw, ᶜK, ᶜts, ᶜp, params) = p
-    (; ᶜinterp) = p.operators
+    (; ᶠuvw, ᶜuvw, ᶜK, ᶜts, ᶜp, params) = p
+    (; ᶠwinterp, ᶜinterp) = p.operators
+    ᶜJ = Fields.local_geometry_field(axes(Y.c.ρ)).J
     C123 = Geometry.Covariant123Vector
-    @. ᶜuvw[colidx] = C123(ᶜuₕ[colidx]) + C123(ᶜinterp(ᶠw[colidx]))
+    @. ᶜuvw[colidx] = C123(ᶜuₕ[colidx]) + C123(ᶜinterp(ᶠw[colidx])) # u̅
+    @. ᶠuvw[colidx] =
```

Comment:
@simonbyrne Here's the suggestion I had for cached `\tilde{u}` : any boundary enforcement would then propagate to the `K` term defined in line 26 at all other points in the code.